sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1798,33 +1817,40 @@ def _encode_y(self, y):
         return encoded_y
 
     def _get_loss(self, sample_weight):
-        if self.loss == "auto":
+        # TODO(1.3): Remove "auto", "binary_crossentropy", "categorical_crossentropy"
+        if self.loss in ("auto", "binary_crossentropy", "categorical_crossentropy"):
+            warnings.warn(
+                f"The loss '{self.loss}' was deprecated in v1.1 and will be removed in "
+                "version 1.3. Use 'log_loss' which is equivalent.",
+                FutureWarning,
+            )
+
+        if self.loss in ("log_loss", "auto"):
             if self.n_trees_per_iteration_ == 1:
-                return _LOSSES["binary_crossentropy"](sample_weight=sample_weight)
+                return HalfBinomialLoss(sample_weight=sample_weight)
             else:
-                return _LOSSES["categorical_crossentropy"](
-                    sample_weight=sample_weight,
-                    n_classes=self.n_trees_per_iteration_,
+                return HalfMultinomialLoss(
+                    sample_weight=sample_weight, n_classes=self.n_trees_per_iteration_
                 )
-
-        if self.loss == "categorical_crossentropy":
+        elif self.loss in ("categorical_crossentropy"):
             if self.n_trees_per_iteration_ == 1:
                 raise ValueError(
-                    "loss='categorical_crossentropy' is not suitable for "
-                    "a binary classification problem. Please use "
-                    "loss='auto' or loss='binary_crossentropy' instead."
+                    f"loss='{self.loss}' is not suitable for a binary classification "
+                    "problem. Please use loss='log_loss' instead."
                 )
             else:
-                return _LOSSES[self.loss](
+                return HalfMultinomialLoss(
                     sample_weight=sample_weight, n_classes=self.n_trees_per_iteration_
                 )
-        else:
+        elif self.loss in ("binary_crossentropy"):
             if self.n_trees_per_iteration_ > 1:
                 raise ValueError(
-                    "loss='binary_crossentropy' is not defined for multiclass"
-                    " classification with n_classes="
-                    f"{self.n_trees_per_iteration_}, use loss="
-                    "'categorical_crossentropy' instead."
+                    f"loss='{self.loss}' is not defined for multiclass "
+                    f"classification with n_classes={self.n_trees_per_iteration_}, "
+                    "use loss='log_loss' instead."
                 )
             else:
-                return _LOSSES[self.loss](sample_weight=sample_weight)
+                return HalfBinomialLoss(sample_weight=sample_weight)
+        else:
+            # should be an instance of BaseLoss
+            return self.loss
```

Comment:
There are no other valid losses, right ?
```suggestion
```

Reply:
I think we should keep this. It enables an unofficial way of specifying one's own loss.
This is a functional unmodified code path. I did not introduce it.

Reply:
I think we should add a test to cover this line:

https://app.codecov.io/gh/scikit-learn/scikit-learn/compare/23040/diff#D1L1854

Reply:
Actually I checked and it's officially privately supported (https://github.com/scikit-learn/scikit-learn/pull/16908/files) and tested. However, the distinction between loss passed as str or BaseLoss is done previously in fit so we can't end up with a custom loss here. Thus this is unreachable code so we can remove it